thunder/tests/opinfos.py

```diff
@@ -2493,6 +2493,54 @@ def where_sample_generator(op, device, dtype, requires_grad, **kwargs):
 conditional_and_mask_ops.append(where_opinfo)
 
 
+def nan_to_num_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    cases = (
+        ((2, 1, 2), None, None, None),
+        ((4, 4), None, 1.0, None),
+        ((4, 4), None, None, 1.0),
+        ((5,), None, 1.0, 0.0),
+        ((8, 1, 6), 1, None, None),
+        ((8, 7, 5, 1), 1, 1.0, None),
+        ((8, 7, 5, 1), 1, None, 0.0),
+        ((8, 7, 5, 1), 1, 1.0, 0.0),
+    )
+
+    for a_shape, nan, posinf, neginf in cases:
```

Comment:
By default, `make_tensor` will only generate values in the interval [-9, 9) for floating point types. See the PyTorch documentation here:

https://pytorch.org/docs/stable/testing.html#torch.testing.make_tensor


When testing `nan_to_num`, this PR should generate tensors by hand (AKA "fixtures") that include NaN, inf, and -inf values for the inexact dtypes. For example:

```
# if the dtype is a floating point type
t = torch.tensor((nan, inf, -inf, inf, ...))
```

```
# if the dtype is a complex type
t = torch.tensor((complex(nan, nan), complex(inf, 0), ...))
```

That way we'll be the sure the function is actually doing something. It might be interesting to look at what `nan_to_num` does with complex values, btw.

Reply:
Wow, can't believe I missed this. Thanks so much! 🚀